template.yml.erb

```diff
@@ -0,0 +1,383 @@
+            SrC/8oPWN4q/IMGA6kiuao+LrqDqyhwoZzKOoeZtn+1xda8v2qLWxpSKmgIdXCeD\n 5mAhk2ybrIXo5vwlizKjCO+zwWECf4uKtWWXBKKqqnz7hZN75E3NX5QQEK6tuzQF\n
+            HCEalY6cg+q12QzaQjV6XM5OV+1fYMwPLNVv1Ocjc/oPpfA+WqPbOd8/OM3bq4f4\n Bp4+XTzk0hDTAgMBAAE=\n
+            -----END RSA PUBLIC KEY-----'
+          rsa_pub_test: '-----BEGIN RSA PUBLIC KEY-----\n MIIBigKCAYEAuZNRSIeapdlDcD0i/lLfw6o7MHTRow5umZyDW8DQ75LcFwbDvOcP\n
+            GD5oSMPuPA6Yc7OYXB08JvHUq1llrxJNP0UGgBwVFZ0Fl1mpzb9Fp0pklw5Q3/tS\n yVH4WCVnFhF0QEFcWJ4WU00DGzrFbVUJw4hHIays0TOmUtCVD4pe6BAz89wf80cy\n
+            SKb+2ABiqC5Qbc9Ki6CLXiOMlXEpG8uYzcmVtFUrA3gyz1Ux+oT+xQ8G2CjL+7Au\n LhBd9Na7zPEsB5sAwHzzZHCPx+/wTKbrT0FfKJRqc0vYOvWFFkeoBcZcx9/+DnGs\n
+            82lnuUHekrW+j3sCimbhFPM+HDXJI03g5Uv698oiSXp8+dqB7VzGD4ux9lwlB6Dr\n 3+8Pz/c7t8me0CJquotr7p7kYtoPo6kK9NL74zu4KsAcotDvRw8tHRs3cZ/X3KEl\n
+            vb2YtSsq4ecf7sZeQgSbeZ6PdwEIBp0dl5AG6SgIWPcweNwmZmJg917i0lSGgWxW\n OlIo9VurlVd3AgMBAAE=\n
+            -----END RSA PUBLIC KEY-----'
+          rsa_pub_adhoc: '-----BEGIN RSA PUBLIC KEY-----\n MIIBCgKCAQEAoGqR0M5qbi+XSH9XCAEQhlEVXbUaIFYXnw0Sxk7YhfAWLpUQMiHW\n
+            DWJemqP54cJ0GPjhHTvL/BdsURkA5ZXpApXWvzQ0kr0kfp7NvC7o2LnB/U78xSsG\n /uwBz+xaCmW71X4uxcXnKT9AFLRRaB1z7RLZoE8NhJErRS+25UqbTBYVdsZkvsIy\n
+            ZUbJMN3hMFIFP4smAtjkKqguLJPFdcMij7WFDjL5kPQpxuc4RSq97iXRODIBjW1B\n zdN9ZpGRA4MDOu3BtLC69PwoFrSULAzlUmC5fJIwpaXoRGZEYRMQThlJ9VsRGwTG\n
+            JTJJ4dWJ2cw61prbeU/HVCWD8OaNumz9tQIDAQAB\n
+            -----END RSA PUBLIC KEY-----'
+  WebSocketAuthorizerPermission:
+    Type: AWS::Lambda::Permission
+    DependsOn:
+      - WebSocketAuthorizer
+    Properties:
+      Action: lambda:InvokeFunction
+      FunctionName: !Ref WebsocketAuthorizerLambda
+      Principal: apigateway.amazonaws.com
+      SourceArn: !Sub "arn:aws:execute-api:${AWS::Region}:${AWS::AccountId}:${WebSocketAPI}/authorizers/${WebSocketAuthorizer}"
+  ChangeJavaRuntimeDirectoryLayer:
+    Type: AWS::Serverless::LayerVersion
+    Properties:
+      CompatibleRuntimes:
+        - java11
+      ContentUri: org-code-javabuilder/change_runtime_directory
+      Description: Change Java runtime to launch from the writeable /tmp directory to enable student projects to write files more easily.
+      LayerName: change-java-runtime-directory
+  FontConfigurationLayer:
+    Type: AWS::Serverless::LayerVersion
+    Properties:
+      CompatibleRuntimes:
+        - java11
+      ContentUri: org-code-javabuilder/font_config.zip
+      Description: Add a font configuration file to enable use of fonts.
+      LayerName: font-configuration
+<%{
+  Theater: {MemorySize: 1769, Timeout: 90},
+  Neighborhood: {MemorySize: 512, Timeout: 60},
+  Console: {MemorySize: 512, Timeout: 60},
+  Playground: {MemorySize: 512, Timeout: 300}
+}.each do |name, config| -%>
+  BuildAndRunJava<%=name%>ProjectFunction:
```

Comment:
One of my minor preferences about cloudformation yaml files is that I like to have a blank line between resources. This is one of those places where that can come in handy because currently I don't immediately know whether I'm still in the context of `FontConfigurationLayer.Properties` until I get to the line after the erb code.